src/main/java/de/westnordost/osmapi/map/changes/MapDataChangesWriter.java

```diff
@@ -149,8 +140,10 @@ private void writeElementAttributes(Element element) throws IOException
 	private void writeNodeContents(Node node) throws IOException
 	{
 		LatLon position = node.getPosition();
-		attribute("lat", position.getLatitude());
-		attribute("lon", position.getLongitude());
+		if (position != null) {
+			attribute("lat", position.getLatitude());
+			attribute("lon", position.getLongitude());
+		}
```

Comment:
The API will reply with an APIBadXMLError (bad request) error if one tries to create or update a node with no position.

So, as having no position at this point is an error anyway, a nullpointer is more descriptive here (and also fails earlier). Would you revert the changes you made for MapDataChangesWriter?